nlp_basics_articles/02_Natural-Language-Processing-Basics.md

```diff
@@ -73,6 +73,72 @@ Figure 6. Q&A
 ## Basic concepts in NLP
 The basic concepts used in NLP are tokens, positional encoding, embedding, and encoder/decoder.
 
-# tokens
-One token of a sequence is generated from the input sequence, representing the smallest language unit.
+### One-hot encoding
```

Comment:
This reminds me of when I worked on the Kaggle Titanic. The cabin numbers don't have any inherent numerical meaning, so treating them as numbers isn't the right approach. One way to take the information into account is to create this kind of flag to indicate cabin positions. This usually increases the dimension of the feature vectors, leading to bad accuracy tho

Reply:
Oh, now that you mentioned it, I can imagine the accuracy would drop. 
Did you have any other way to represent the cabin numbers? Maybe some compressed version?